packages/components/src/Payments/PaymentsEth.tsx

```diff
@@ -55,12 +58,22 @@ export class PaymentsEth extends React.Component<PaymentsEthProps, PaymentsEthSt
     };
   }
 
+  public async componentDidMount(): Promise<void> {
+    // Grab user address after any `currentProviderEnable` call. For users who aren't logged in to Civil but are logged in to metamask, this is the only way we can get their address:
+    this.setState({
+      userAddress: await this.context.civil.accountStream.first().toPromise(),
+    });
+  }
+
   public render(): JSX.Element {
-    if (!this.props.isWalletConnected) {
+    const userAddress =
+      this.state.userAddress || (this.context && this.context.currentUser && this.context.currentUser.ethAddress);
+
+    if (!userAddress) {
       return (
         <PaymentsEthWrapper etherToSpend={this.state.etherToSpend} usdToSpend={this.state.usdToSpend}>
           <ConnectWalletWarningText />
-          <PaymentBtn>Select Wallet</PaymentBtn>
+          <PaymentBtn onClick={() => this.context.civil.currentProviderEnable()}>Select Wallet</PaymentBtn>
```

Comment:
can we update this button (in a later PR is fine) so that it becomes disabled / adds a spinner while this async call is being completed?

Reply:
If a user has previously connected wallet and is logged in to wallet but for some reason eth isn't enabled (happens) then this call is basically instantaneous. If MM (haven't tested portis) isn't connected and/or not unlocked then it pops up MM. If the user closed or didn't see MM popup and we put a disabled loading state on button, they'd be kind of stuck. As it is now, they could at least click button again and maybe get popup again? Not sure what the best UX is here.